### DISCUSSION TITLE
403 Forbidden, even in a valid Geographical jurisdiction.

### DISCUSSION THREAD

**Discussion by Ro-Knight** [cloudbet/docs#43] [Q&A] [answered]
Hi CloudBet team,

We have been trying to deploy our code in GCP - App Engine, and in a specific region that is not part of the Geographical restrictions stated in here [Cloudbet Terms & Conditions](https://www.cloudbet.com/en/help/terms). For example we have tried to deploy it in Montreal Region (Canada) and we still get a 403 Forbidden, we have tried it in Australia and same result. 

When we try it locally from our machine, the request goes through (based in Canada)

Would you know why else it would give a 403 Forbidden? Note that this only happens for when placing a bet with the Trading api.

Any insight would help. Thanks!

**Comment by kgravenreuth** [comment, marked as answer]
Hi @Ro-Knight 

Thanks for providing the details on this issue. 

In our experience with other customers, such issues might happen with Cloud Providers such as GCP if you are using auto-assigned IP addresses which might be taken from a pool of IP addresses detected as being from a different geography than the one you provisioned your service in.  For example, if you spin up a service in Canada but the auto-assigned IP address could be detected as being from a restricted geography such as the USA.

We would recommend you to follow your cloud provider's instructions on provisioning a static IP address explicitly in the geography where your service is provisioned. For example, please refer to the [Documentation for GCP App Engine](https://cloud.google.com/appengine/docs/legacy/standard/python/outbound-ip-addresses#static-ip)

Let me know if this addresses your issue or if you need any additional help here. 

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by Ro-Knight** [reply]
Sorry for the late reply on this one @kgravenreuth . Thanks for the guidance and that actually worked, we had a little issue connecting to our Database after we had configure that static ip address hence my late response. But then all got fixed and we are able to reach Cloudbet from the Google App Engine instance. Thanks again for your quick response and guidance!
